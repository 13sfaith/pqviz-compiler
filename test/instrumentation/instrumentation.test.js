import runInstrumentation from '../../instrumentation/run-instrumentation.js'
import { expect, test } from 'vitest'
import babel from "@babel/core"
import path from 'path'
import fs from 'fs'

function getTestInput(codeExampleName) {
    const fileName = `test/instrumentation/inputs/${codeExampleName}.js`
    const originalCode = fs.readFileSync(path.join(process.cwd(), fileName), "utf8");
    return originalCode
}

function instrumentTestInput(codeExampleName) {
    const originalCode = getTestInput(codeExampleName)
    const newCode = runInstrumentation(originalCode, codeExampleName)
    return newCode
}

test('Standard Functions Get Instrumented', () => {
    const instrumentedCode = instrumentTestInput('standard-function-definitions')
    expect(instrumentedCode).toMatchSnapshot()
})