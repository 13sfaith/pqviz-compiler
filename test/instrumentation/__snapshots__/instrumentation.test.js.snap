// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Standard Functions Get Instrumented 1`] = `
"import pqvizMonitor from "./pqviz/monitor.js";
pqvizMonitor.addEvent({
  "type": "moduleStart",
  "file": "standard-function-definitions"
})
function foo() {
  pqvizMonitor.addEvent({
    "type": "functionStart",
    "name": "foo",
    "file": "standard-function-definitions",
    "line": 1
  });
  let a = 10;
  (() => {
    pqvizMonitor.addEvent({
      "type": "functionCall",
      "from": "foo",
      "to": "bar",
      "callingFile": "standard-function-definitions",
      "callingLine": 3,
      "args": ["Unimplemented Literal Type: object"]
    });
    let pqvizMonitorTmp = bar(a);
    pqvizMonitor.addEvent({
      "type": "functionReturn",
      "from": "foo",
      "to": "bar",
      "callingFile": "standard-function-definitions",
      "callingLine": 3
    });
    return pqvizMonitorTmp;
  })();
}
async function bar(val) {
  pqvizMonitor.addEvent({
    "type": "functionStart",
    "name": "bar",
    "file": "standard-function-definitions",
    "line": 6
  });
  console.log("hello: ", val);
}
(() => {
  pqvizMonitor.addEvent({
    "type": "functionCall",
    "from": "TLS",
    "to": "foo",
    "callingFile": "standard-function-definitions",
    "callingLine": 10,
    "args": []
  });
  let pqvizMonitorTmp = foo();
  pqvizMonitor.addEvent({
    "type": "functionReturn",
    "from": "TLS",
    "to": "foo",
    "callingFile": "standard-function-definitions",
    "callingLine": 10
  });
  return pqvizMonitorTmp;
})();"
`;
